src/passes/rejectUnsupportedFeatures.ts

```diff
@@ -93,6 +97,33 @@ export class RejectUnsupportedFeatures extends ASTMapper {
     this.addUnsupported('User defined Errors are not supported', node);
   }
 
+  visitEventDefinition(node: EventDefinition, ast: AST): void {
+    node.vParameters.vParameters.forEach((param) => {
+      if (param.indexed) {
+        const paramType = generalizeType(safeGetNodeType(param, ast.inference))[0];
+        if (isValueType(paramType)) {
+          this.commonVisit(node, ast);
+          return;
+        }
+        if (
+          paramType instanceof ArrayType ||
+          paramType instanceof MappingType ||
+          (paramType instanceof PointerType && isReferenceType(paramType.to))
```

Comment:
Why no using `isReferenceType` instead of the boolean expression?

Reply:
string/bytes types are supported